### PR TITLE
feat(utilities): Add scss utility functions to find fixed and fluid sizes.

### LINKED ISSUE #21: Transform column and unit finding into functions
@jendowns, are you down to move those functions you worked on internally into Gridish?

### PR DESCRIPTION
**Pull request by jendowns**
## Summary 

Resolves #21 
This PR adds utility functions to helps consuming developers calculate and use fixed and fluid sizes within their scss. Notably, the fluid sizing is based on the breakpoint values that the consuming developer set in their `css-gridish.json`. 

An example use case would be a consuming developer using `getFluidSize()` to set a component's width to be a fluid size, relative to their grid configuration (see example below in the "Test" section).

## Proposed Changes

* **Adds `bin/src/scss/_utilities.scss`**, which contains the following utility functions:
  * `map-deep-get` - for traversing deeply nested maps like the generated `$grid-values` map
  * `getBreakpointValue` - returns a breakpoint value in rems when passed a breakpoint name (example use: `@media` queries)
  * `getFluidSize` - returns a `calc()` expression that represents a fluid length (example use: `min-width` rule)
  * `getFixedSize` - returns a length in rems when passed a fixed nondimensional unit
* **Updates Examples (scss & Sketch files) after running `npm run build-examples`**

## Testing with the Examples

**I suggest testing with the examples _individually_**. If you try to test them all at once by adding some sample rules that use `getFluidSize()` or `getBreakpointValue()`, you will run into trouble because the Material example is using different breakpoint names ("small" instead of "sm") & the `map-get()` or `map-deep-get()` in each of those functions won't be able to find the breakpoint names you are using. If you test them all individually, you can modify the breakpoints names in your test scss rules as appropriate.

### Bootstrap & Carbon

1.  Add the following the bottom of the new `bin/src/scss/_utilities.scss`:

```
@media screen and (min-width: getBreakpointValue('sm')){
  .awesome-test-element {
    max-width: getFixedSize(10);
    min-width: getFluidSize('sm', 1);
  }
}
```

2. `cd examples/bootstrap`

3. `node ../../bin/index.js` to run just the Bootstrap example

4. view the compiled CSS and check for `.awesome-test-element`

5. repeat for Carbon

### Material

The Material `css-gridish.json` is using different breakpoint names ("small" instead of "sm", for example). So the steps to test are the same as above, but modify the breakpoint names to match the full words in that example's `css-gridish.json` file:

```
@media screen and (min-width: getBreakpointValue('small')){
  .awesome-test-element {
    max-width: getFixedSize(10);
    min-width: getFluidSize('small', 1);
  }
}
```

### CODE REVIEW

**Comment by seejamescode** [comment]
Thanks Jen! Can you merge latest master and see where we moved the src directory?

File: src/scss/_utilities.scss [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+/// @author Hugo Giraudel
+/// @param {Map} $map A sass map and any number of keys.
+/// @param {*} $keys Nested values.
+/// @return Nested values or nested map.
+@function map-deep-get($map, $keys...) {
+  @each $key in $keys {
+    $map: map-get($map, $key);
+  }
+  @return $map;
+}
```
Comment (seejamescode) [inline src/scss/_utilities.scss]: - [ ] `map-deep-get` should move to `_functions.scss`

File: src/scss/_utilities.scss [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+/// Function returns the value for a given breakpoint.
+/// @param {String} $breakpointName Breakpoint name (sm, md, lg, xlg, max).
+/// @return {Length} The breakpoint value in rems.
+@function getBreakpointValue($breakpointName) {
```
Comment (seejamescode) [inline src/scss/_utilities.scss]: - [ ] I think @diego-codes wants us to move to kebab-case on these

File: src/scss/_utilities.scss [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  $breakpointValue: map-deep-get($allBreakpoints, $breakpointName, 'breakpoint');
+  
+  @return $breakpointValue * 1rem;
+}
+
+/// Returns a calc() expression that represents a fluid width at a given breakpoint.
+/// @param {String} $breakpointName A valid breakpoint (sm, md, lg, xlg, or max).
```
Comment (seejamescode) [inline src/scss/_utilities.scss]: Do not list specific names here since they will differ for different users

File: src/scss/_utilities.scss (hunk 2 above)
Comment (seejamescode) [reply src/scss/_utilities.scss]: It is okay if you want to knock it out in the same one. I changed the base of the PR to v1.1.0.

Comment (jendowns) [reply src/scss/_utilities.scss]: So you'd like to see me update everything in `functions.scss` & `mixins.scss`, or just in this new `utilities.scss` file? (If the latter, I worry about there still being inconsistency, as I'm using variables like `$allBreakpoints` in camel case, which come from other scss files).

Comment (seejamescode) [reply src/scss/_utilities.scss]: I just mean function and mixin names 😄

**Pull request merged**